raiden/tests/integration/rpc/assumptions/test_rpc_gas_estimation_assumptions.py

```diff
@@ -77,3 +80,41 @@ def test_estimate_gas_defaults_to_pending(deploy_client):
     assert second_receipt["gasLimit"] < deploy_client.get_block("latest")["gasLimit"]
     assert first_receipt["status"] != RECEIPT_FAILURE_CODE
     assert second_receipt["status"] != RECEIPT_FAILURE_CODE
+
+
+def test_estimate_gas_for_dependent_transactions_needs_a_mined_transaction(
+    deploy_client: JSONRPCClient,
+) -> None:
+    """Gas estimation for a transaction that depends on another works afte
+    eth_getTransaction returns a valid value.
+
+    This test makes sure that consecutive transactions which depends on the
+    changes from pending ones can have their gas estimate after
+    `eth_getTransaction` returns. This assumption is important for the fast
+    handling of `approve` and `setTotalDeposit` transactions, where the
+    `setTotalDeposit` needs the result of an estimate_gas just after sending an
+    approve transaction.
+    """
+    contract_proxy, _ = deploy_rpc_test_contract(deploy_client, "RpcWithStorageTest")
+
+    next_counter = 0
+    iterations = 100
+
+    # Do enough transactions so that they won't fit in a single block.
+    for _ in range(20):
+        next_counter += 1
+
+        gaslimit = contract_proxy.estimate_gas("pending", "next", next_counter, iterations)
+        msg = (
+            "gas estimation should not have failed, this means the side-effects "
+            "of the previous transaction have not been accounted for and the "
+            "strategy of polling is not sufficient to avoid race conditins."
+        )
+        assert gaslimit, msg
+
+        tx = contract_proxy.transact("next", gaslimit, next_counter, iterations)
+
+        # Neither `eth_getTransaction` and `eth_pendingTransactions` are
+        # sufficient here, it still possible to have race conditoins with both
+        # of these RPC calls.
+        deploy_client.poll(tx)
```

Comment:
Initially I wrote the test to make sure that `eth_getTransaction` would be sufficient here ... turns out it is not. the only solution I have found is to wait for the transaction to be polled and confirmed.